### PR TITLE
Issue #7247: Deprecate SelectionAwareSessionObserver

### PR DESCRIPTION
**Pull request by csadilek**
With recent refactorings, we don't use it in Fenix anymore. Follow-up tickets are filed for remaining features and linked in source.

### CODE REVIEW

**Review by pocmo — APPROVED** [review]
🚀

File: samples/browser/src/main/java/org/mozilla/samples/browser/BaseBrowserFragment.kt [hunk 1]
```diff
@@ -209,13 +196,28 @@ abstract class BaseBrowserFragment : Fragment(), UserInteractionHandler {
...
             secureWindowFeature
         )
 
         return layout
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        consumeFlow(components.store) { flow ->
+            flow.mapNotNull { state -> state.findCustomTabOrSelectedTab(sessionId) }
+                .ifAnyChanged { tab ->
```
Comment (Amejia481) [inline samples/browser/src/main/java/org/mozilla/samples/browser/BaseBrowserFragment.kt:208]: Should we add a test to verify that we are observing the store when `onViewCreated`?

**Review by Amejia481 — APPROVED** [review]
Looks good, I just have one test question, but not blocking.

File: samples/browser/src/main/java/org/mozilla/samples/browser/BaseBrowserFragment.kt (hunk 1 above)
Comment (csadilek) [reply samples/browser/src/main/java/org/mozilla/samples/browser/BaseBrowserFragment.kt:208]: This is Sample Browser.

Comment (csadilek) [reply samples/browser/src/main/java/org/mozilla/samples/browser/BaseBrowserFragment.kt:208]: We're not unit testing / just UI testing our Sample apps. I don't think we want to change that?

Comment (csadilek) [reply samples/browser/src/main/java/org/mozilla/samples/browser/BaseBrowserFragment.kt:208]: (But we do want more UI tests in sample browser for sure) :)

**Pull request merged**